### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by mquandalle** [s-yadav/react-number-format#19]
Hello,

It is currently impossible to import NumberFormat in the ES6 standard manner when using Typescript:

```ts
// Doesn't work
import NumberFormat from "react-number-format"

// Workaround
import NumberFormat = require("react-number-format")
```

It would also be great to include a typing declaration file for automatic auto-completion in IDEs. The following snippet should be a good basis: 

```ts
declare module 'react-number-format' {
  export class NumberFormat extends React.Component<NumberFormatProps, any> {
    // Any public method on the NumberFormat component?
  }

  export interface NumberFormatProps extends React.HTMLAttributes<HTMLInputElement> {
    thousandSeparator?: boolean | ',' | '.' | ' '
    decimalSeparator?: boolean | ',' | '.'
    displayType?: 'input' | 'text'
    prefix?: string
    format?: string | Function
    mask?: string
    // value?: number | string
  }

  export default NumberFormat;
}
```

**Comment by s-yadav** [comment]
Strange, in babel it works. It should work on typescript as well. Never tried though in ts. Will check and see what can be done for this.

**Comment by danwoodbury** [comment]
i got it working with typescript using:

`import * as NumberFormat from 'react-number-format';`

**Comment by s-yadav** [comment]
Added it on readme.

**Comment by Mario-Eis** [comment]
`import * as NumberFormat from 'react-number-format';` only works if "implicit any" is allowed for the project. I would not suggest such a configuration for professional projects.

**Comment by horacehylee** [comment]
To whom it may concern, the proper way of importing `react-number-format` should be
```typescript
import { default as NumberFormat } from 'react-number-format';
```
which then can set `noImplicitAny` to `true` and without having to use `require` to import the module.

Moreover, my [pull request](https://github.com/s-yadav/react-number-format/pull/200) has been merged, so declaration file will be available by default for using Typescript after `v3.4.4`.
